flux-middle/src/rty/expr.rs

```diff
@@ -267,6 +267,26 @@ impl Expr {
         matches!(self.kind, ExprKind::BinaryOp(..))
     }
 
+    fn const_op(op: &BinOp, c1: &Constant, c2: &Constant) -> Option<Constant> {
+        match op {
+            BinOp::Iff => c1.iff(c2),
+            BinOp::Imp => c1.imp(c2),
+            BinOp::Or => c1.or(c2),
+            BinOp::And => c1.and(c2),
+            BinOp::Gt => c1.gt(c2),
+            BinOp::Ge => c1.ge(c2),
+            BinOp::Lt => c2.gt(c1),
+            BinOp::Le => c2.ge(c1),
+            BinOp::Add => c1.add(c2),
+            BinOp::Sub => c1.sub(c2),
+            BinOp::Mul => c1.mul(c2),
+            BinOp::Div => c1.div(c2),
+            BinOp::Mod => c1.mod_(c2),
+            BinOp::Eq => Some(c1.eq(c2)),
+            BinOp::Ne => Some(c1.ne(c2)),
+        }
+    }
+
     /// Simplify expression applying some simple rules like removing double negation. This is
     /// only used for pretty printing.
     pub fn simplify(&self) -> Expr {
```

Comment:
Note that this function is only used for pretty printing and it won't affect the constraint we throw at fixpoint.

Reply:
You probably read that comment in `Expr::is_trivially_true`.

Yeah, comparisons should be easier than general constant folding. We could call `Expr::simplify` [here](https://github.com/liquid-rust/flux/blob/9bf0d9cc21fb2998c7abaa24b09d8b3316b2da24/flux-refineck/src/refine_tree.rs#L382). But note that we still need to implement comparisons in the signs + usize representation, but I think if we define it as a struct with the fields in the right order then we could `#[derive(PartialOrd, Ord, PartialEq, Eq)]`. The alternative is to extend `Expr::is_trivially_true` to look for trivial comparisons. The case `1=1` is already handled there for example.